spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRewriteDataFilesProcedure.java

```diff
@@ -356,6 +356,16 @@ public void testRewriteDataFilesWithInvalidInputs() {
                     + "sort_order => 'c1 ASC NULLS FIRST')",
                 catalogName, tableIdent));
 
+    // Test for sort strategy without any (default/user defined) sort_order
+    AssertHelpers.assertThrows(
+        "Should reject calls with error message",
+        IllegalArgumentException.class,
+        "Can't use SORT when there is no sort order",
```

Comment:
"when the table has no defined sort order and the user has not set a specific sort order"?

Reply:
this is the current full error message `Can't use %s when there is no sort order, either define table %s's sort order or set sort order in the action`